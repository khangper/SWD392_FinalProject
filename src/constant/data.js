export const API_KEY = "API_KEY";

export const FETCH_HOME_LIVESTREAM_REQUEST = "FETCH_HOME_LIVESTREAM_REQUEST";
export const FETCH_HOME_LIVESTREAM_SUCCESS = "FETCH_HOME_LIVESTREAM_SUCCESS";
export const FETCH_HOME_LIVESTREAM_FAILURE = "FETCH_HOME_LIVESTREAM_FAILURE";

export const FETCH_HOME_FEATURED_COURSES_REQUEST =
  "FETCH_HOME_FEATURED_COURSES_REQUEST";
export const FETCH_HOME_FEATURED_COURSES_SUCCESS =
  "FETCH_HOME_FEATURED_COURSES_SUCCESS";
export const FETCH_HOME_FEATURED_COURSES_FAILURE =
  "FETCH_HOME_FEATURED_COURSES_FAILURE";
// All Live Stream
export const FETCH_STREAMS_REQUEST = "FETCH_STREAMS_REQUEST";
export const FETCH_STREAMS_SUCCESS = "FETCH_STREAMS_SUCCESS";
export const FETCH_STREAMS_FAILURE = "FETCH_STREAMS_FAILURE";

//All Instructor
export const FETCH_INSTRUCTORS_REQUEST = "FETCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTORS_SUCCESS = "FETCH_INSTRUCTORS_SUCCESS";
export const FETCH_INSTRUCTORS_FAILURE = "FETCH_INSTRUCTORS_FAILURE";
export const SEARCH_INSTRUCTORS_REQUEST = "SEARCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTOR_BY_ID_REQUEST = "FETCH_INSTRUCTOR_BY_ID_REQUEST";

// SaveCourse.js
export const FETCH_SAVECOURSES_REQUEST = "FETCH_SAVECOURSES_REQUEST";
export const FETCH_SAVECOURSES_SUCCESS = "FETCH_SAVECOURSES_SUCCESS";
export const FETCH_SAVECOURSES_FAILURE = "FETCH_SAVECOURSES_FAILURE";
export const DELETE_SAVECOURSE_REQUEST = "DELETE_SAVECOURSE_REQUEST";
export const DELETE_SAVECOURSE_SUCCESS = "DELETE_SAVECOURSE_SUCCESS";
export const DELETE_SAVECOURSE_FAILURE = "DELETE_SAVECOURSE_FAILURE";

// Shoppingcart.js
export const ShoppingCart_API_URL =
  "https://66908916c0a7969efd9c67ed.mockapi.io/ojt-repo/ShoppingCart";
export const ADD_TO_CART = "ADD_TO_CART";
export const REMOVE_FROM_CART = "REMOVE_FROM_CART";
export const APPLY_DISCOUNT = "APPLY_DISCOUNT";
export const ADD_TO_CART_FAILURE = " ADD_TO_CART_FAILURE";
export const ADD_TO_CART_SUCCESS = "APPLY_DISCOUNT";

//MyCourse_Course
export const FETCH_COURSES_REQUEST = "FETCH_COURSES_REQUEST";
export const FETCH_COURSES_SUCCESS = "FETCH_COURSES_SUCCESS";
export const FETCH_COURSES_FAILURE = "FETCH_COURSES_FAILURE";

export const EDIT_COURSE_REQUEST = "EDIT_COURSE_REQUEST";
export const EDIT_COURSE_SUCCESS = "EDIT_COURSE_SUCCESS";
export const EDIT_COURSE_FAILURE = "EDIT_COURSE_FAILURE";

export const DELETE_COURSE_REQUEST = "DELETE_COURSE_REQUEST";
export const DELETE_COURSE_SUCCESS = "DELETE_COURSE_SUCCESS";
export const DELETE_COURSE_FAILURE = "DELETE_COURSE_FAILURE";

//UpcommingCourse_Course
export const FETCH_UPCOMMINGCOURSES_REQUEST = "FETCH_UPCOMMINGCOURSES_REQUEST";
export const FETCH_UPCOMMINGCOURSES_SUCCESS = "FETCH_UPCOMMINGCOURSES_SUCCESS";
export const FETCH_UPCOMMINGCOURSES_FAILURE = "FETCH_UPCOMMINGCOURSES_FAILURE";

export const EDIT_UPCOMMINGCOURSE_REQUEST = "EDIT_UPCOMMINGCOURSE_REQUEST";
export const EDIT_UPCOMMINGCOURSE_SUCCESS = "EDIT_UPCOMMINGCOURSE_SUCCESS";
export const EDIT_UPCOMMINGCOURSE_FAILURE = "EDIT_UPCOMMINGCOURSE_FAILURE";

export const DELETE_UPCOMMINGCOURSE_REQUEST = "DELETE_UPCOMMINGCOURSE_REQUEST";
export const DELETE_UPCOMMINGCOURSE_SUCCESS = "DELETE_UPCOMMINGCOURSE_SUCCESS";
export const DELETE_UPCOMMINGCOURSE_FAILURE = "DELETE_UPCOMMINGCOURSE_FAILURE";
