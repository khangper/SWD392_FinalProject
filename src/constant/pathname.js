export const PATH_NAME = {
  HOME: "/home",
  EXPLORE: "/explore",
  LIVE_OUTPUT: "/live_output/:id",
  ADD_LIVE_STREAM: "/add_livestream",
  ALL_LIVE_STREAM: "/all_livestream",
  ALL_INSTRUCTOR_LIST: "/allinstructorlist/:id",
  HELP: "/help",
  REPORT_HISTORY: "/reporthistory",
  SEND_FEEDBACK: "/sendfeedback",
  OTHER_INSTRUCTOR_VIEW: "/other-instructor-view/:id",
  MY_INSTRUCTOR_PROFILE: "/my-instructor-profile",
  SAVE_COURSES: "/save-courses",
  SETTING: "/setting",
  MY_COURSE: "/courses",
  LOGIN: "/login",
  FORGOT_PASSWORD: "/forgot-password",
  SIGN_UP: "/sign-up",
  SIGN_UP_STEP: "/SignUpStep",
  THANKS: "/thanks",
  ERROR_404: "/error404",
  INVOICE: "/invoice",
  PRESS: "/press",
  SEARCH_RESULT: "/search_result",
  PAID_MEMBERSHIP: "/paidmembership",
  ABOUT_US: "/about-us",
  OUR_BLOG: "/our-blog",
  BLOG_SINGLE_VIEW: "/blog-single-view",
  CHECKOUT: "/checkout",
  SHOPPING_CART: "/shoppingcart",
  HELP_VIEW_1: "/helpview1",
  HELP_VIEW_2: "/helpview2",
  COMPANY_DETAILS: "/company-details",
  CAREER: "/career",
  APPLY_JOB: "/apply-job",
  CONTACT_US: "/contactus",
  TERMS_OF_USE: "/termsofuse",
  CERTIFICATION_FILL_FORM: "/certificationfillform",
  CERTIFICATION_TEST: "/certificates-test",
  TEST_RESULT: "/test-result",
  MESSAGE: "/messages",
  COURSES_DETAIL_VIEW: "/courses-detail-view/:id",
  PREVIEWS: "/reviews",
  COMING_SOON: "/coming-soon",
  INS_DASHBOARD: "/ins-dashboard",
  PURCHASE_COURSE: "/PurchaseCourse",
  NOTIFICATON: "/notifications",
  DOWLOADCOURSE: "/dowloadcourse",
  CERTIFICATION_CENTER: "/certificationcenter",
  SITEMAP: "/sitemap",
  EARNING: "/earning",
  PAYOUT: "/payout",
  VERTIFICATION: "/vertification",
  MYCERTIFICATIONS: "/mycertifications",
  CREDIT: "/credits",
  STATEMENTIS: "/statementis",
  STATEMENTST: "/statementst",
  ANALYTICS: "/analytics",
  CREATE_NEW_COURSE: "/create-course",
  STD_DASHBOARD: "/student-dashboard",
  NOTIFICATIONSTUDENT: "/notificationsstudent",
};
