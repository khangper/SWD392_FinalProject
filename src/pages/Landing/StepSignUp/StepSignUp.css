/* StepSignUp.css */
.tablist {
  display: flex;
  justify-content: center; /* Canh giữa các tab */
  align-items: center;
}

.tab {
  cursor: pointer;
  padding: 10px 20px;
  border-bottom: 2px solid transparent;
  margin: 0 5px; /* Giảm khoảng cách giữa các tab */
}

.tab.active {
  border-bottom-color: #196c7e; /* Màu sắc cho tab active */
}

.tab .instructor-sign-up,
.tab .student-sign-up {
  font-size: 18px;
  font-weight: bold;
}

  /* Thêm CSS để tạo hiệu ứng hoặc màu sắc khác cho tab active nếu cần */
.login-bannerInstructor{
    color: #666;
    margin: 0;
    margin-bottom: 10px;
    margin-top: 37px;
    margin-bottom: 56px;
}  

.selectpicker {
    width: 80%;
    padding: 10px;
    color: #666;
    border: 1px solid #ddd;
    border-radius: 4px;
    font-size: 14px;
    padding: 10px; /* Adjust padding to match input */
    cursor: pointer; /* Show pointer cursor on hover */
}

.selectpicker option.blue {
    color: #196c7e;
  }

  .selectpicker option {
    background-color: white;
    padding: 40px; /* Increase padding for larger option size */
    font-size: 20px; /* Increase font size */
  }
  .selectpicker option:hover {
    background-color: yellow;

  }

  .textarea {
    margin-top: 22px;
    width: 80%;
    position: relative;
    box-shadow: 0px 1px 4px 1px rgba(0, 0, 0, 0.02) inset;
    border-radius: 3px;
    background-color: #fff;
    border: 1px solid #e5e5e5;
    box-sizing: border-box;
    height: 77.9px;
    overflow: hidden;
    }

    .textarea1{
      margin-top: -18px;
      width: 80%;
      position: relative;
      box-shadow: 0px 1px 4px 1px rgba(0, 0, 0, 0.02) inset;
      border-radius: 3px;
      background-color: #fff;
      border: 1px solid #e5e5e5;
      box-sizing: border-box;
      height: 77.9px;
      overflow: hidden;
    }

    .textarea1-student {
      margin-top: -18px;
      width: 80%;
      position: relative;
      box-shadow: 0px 1px 4px 1px rgba(0, 0, 0, 0.02) inset;
      border-radius: 3px;
      background-color: #fff;
      border: 1px solid #e5e5e5;
      box-sizing: border-box;
      height: 158.9px;
      overflow: hidden;
  }

    .help-block {
      width: 390.9px;
      position: relative;
      font-size: 12px;
      line-height: 20px;
      font-family: Roboto;
      color: #686f7a;
    }
    
    .top38px{
      margin-top: 38px;
    }
  
    .ui.form .singnupstep-field {
      margin-right: -8px;
  }

  .textarea {
    padding: 10px; /* Adjust padding as needed */
    border: 1px solid #ccc;
    border-radius: 4px;
    width: 100%;
    box-sizing: border-box; /* Ensure padding doesn't affect the width */
    font-size: 14px; /* Adjust font size as needed */
    resize: none; /* Prevents resizing, optional */
  }