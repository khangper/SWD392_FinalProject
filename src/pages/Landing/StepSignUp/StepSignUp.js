import React, { useState } from "react";
import { Link } from "react-router-dom";
import "./StepSignUp.css";
import { PATH_NAME } from "../../../constant/pathname";

export default function StepSignUp() {
  const [activeTab, setActiveTab] = useState("instructor"); // State to store the active tab
  const [selectedOption, setSelectedOption] = useState(""); // State to store the selected option
  const [description, setDescription] = useState(""); // State to store the description
  const [warningMessage, setWarningMessage] = useState(""); // State to store the warning message

  const handleTabChange = (tab) => {
    setActiveTab(tab);
  };

  const handleSelectChange = (event) => {
    setSelectedOption(event.target.value);
  };

  const handleInputChange = (event) => {
    const value = event.target.value;
    setDescription(value);

    const specialCharPattern = /[^a-zA-Z0-9\s.,?!]/;
    if (specialCharPattern.test(value)) {
      setWarningMessage("Special characters are not allowed.");
    } else {
      setWarningMessage("");
    }
  };

  return (
    <div className="login-container">
      <div className="login-logo-container">
        <Link to={PATH_NAME.HOME}>
          <img
            src="https://gambolthemes.net/html-items/cursus-new-demo/images/logo.svg"
            alt="Cursus Logo"
            className="login-logo"
          />
        </Link>
      </div>
      <div className="login-form-container">
        <div className="tablist">
          <div
            className={`tab ${activeTab === "instructor" ? "active" : ""}`}
            onClick={() => handleTabChange("instructor")}
          >
            <div className="instructor-sign-up">Instructor Sign Up</div>
          </div>
          <div
            className={`tab ${activeTab === "student" ? "active" : ""}`}
            onClick={() => handleTabChange("student")}
          >
            <div className="student-sign-up">Student Sign Up</div>
          </div>
        </div>
        {activeTab === "instructor" && (
          <div className="instructor-sign-up-form">
            <p className="login-bannerInstructor">Sign Up and Create Course!</p>
            <form>
              <select
                className="selectpicker"
                onChange={handleSelectChange}
                value={selectedOption}
              >
                <option value="">Select Category</option>
                <option value="1">Development</option>
                <option value="2">Business</option>
                <option value="3">Finance & Accounting</option>
                <option value="4">IT & Software</option>
                <option value="5">Office Productivity</option>
                <option value="6">Personal Development</option>
                <option value="7">Design</option>
                <option value="8">Marketing</option>
                <option value="9">Lifestyle</option>
                <option value="10">Photography</option>
                <option value="11">Health & Fitness</option>
                <option value="12">Music</option>
                <option value="13">Teaching & Academics</option>
              </select>
              <div className="ui search focus mt-15">
                <div className="ui form swdh30">
                  <div className="singnupstep-field">
                    <textarea
                      rows="3"
                      name="description"
                      id="id_about"
                      placeholder="Write a little description about you..."
                      className="textarea"
                      value={description}
                      onChange={handleInputChange}
                    ></textarea>
                  </div>
                </div>
                {warningMessage && (
                  <div className="help-block special-char-warning">
                    {warningMessage}
                  </div>
                )}
                <div className="help-block">
                  Your biography should have at least 12000 characters.
                </div>
              </div>
              <button className="loginSingUp-btn" type="submit">
                Instructor Sign Up Now
              </button>
            </form>
            <p className="top38px">
              Already have an account?{" "}
              <Link to="/Login" className="LoginMove">
                Log In
              </Link>
            </p>
          </div>
        )}
        {activeTab === "student" && (
          <div className="student-sign-up-form">
            <div className="instructor-sign-up-form">
              <p className="login-bannerInstructor">
                Sign Up and Start Learning
              </p>
              <form>
                <div className="ui search focus mt-15">
                  <div className="ui form swdh30">
                    <div className="singnupstep-field">
                      <textarea
                        rows="3"
                        name="description"
                        id="id_about"
                        placeholder="Write a little description about you..."
                        className="textarea"
                        value={description}
                        onChange={handleInputChange}
                      ></textarea>
                    </div>
                  </div>
                  {warningMessage && (
                    <div className="help-block special-char-warning">
                      {warningMessage}
                    </div>
                  )}
                  <div className="help-block">
                    Your biography should have at least 12000 characters.
                  </div>
                </div>
                <button className="loginSingUp-btn" type="submit">
                  Student Sign Up Now
                </button>
              </form>
              <p className="top38px">
                Already have an account?{" "}
                <Link to="/Login" className="LoginMove">
                  Log In
                </Link>
              </p>
            </div>
          </div>
        )}
      </div>
      <div className="form-footer">
        <img
          alt=""
          className="footer-logo"
          src="https://gambolthemes.net/html-items/cursus-new-demo/images/sign_logo.png"
        />
        <span>
          Â© 2024 <strong>Cursus</strong>. All Rights Reserved.
        </span>
      </div>
    </div>
  );
}
