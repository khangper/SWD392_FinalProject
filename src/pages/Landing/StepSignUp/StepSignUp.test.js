import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import StepSignUp from "../../Landing/StepSignUp/StepSignUp";
import "@testing-library/jest-dom/extend-expect";
import { renderWithProviders } from "../../../constant/test-util";

describe("StepSignUp", () => {
  beforeAll(() => {
    jest.spyOn(console, "error").mockImplementation(() => {});
    jest.spyOn(console, "log").mockImplementation(() => {});
  });

  afterAll(() => {
    console.error.mockRestore();
    console.log.mockRestore();
  });

  it("renders the StepSignUp component", () => {
    const { container } = renderWithProviders(<StepSignUp />);
    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();
  });

  it("switches to student tab when clicked", () => {
    renderWithProviders(<StepSignUp />);
    const studentTab = screen.getByText("Student Sign Up");
    fireEvent.click(studentTab);
    expect(studentTab.parentElement).toHaveClass("active");
    const studentForm = screen.getByText("Sign Up and Start Learning");
    expect(studentForm).toBeInTheDocument();
  });

  it("updates selectedOption state when a category is selected", () => {
    renderWithProviders(<StepSignUp />);
    const selectElement = screen.getByRole("combobox");
    fireEvent.change(selectElement, { target: { value: "2" } });
    expect(selectElement.value).toBe("2");
  });

  it("displays warning message when special characters are entered in the description", () => {
    renderWithProviders(<StepSignUp />);
    const textarea = screen.getByPlaceholderText(
      "Write a little description about you..."
    );
    fireEvent.change(textarea, { target: { value: "Hello@!" } });
    const warningMessage = screen.getByText(
      "Special characters are not allowed."
    );
    expect(warningMessage).toBeInTheDocument();
  });

  it("removes warning message when special characters are removed from description", () => {
    renderWithProviders(<StepSignUp />);
    const textarea = screen.getByPlaceholderText(
      "Write a little description about you..."
    );
    fireEvent.change(textarea, { target: { value: "Hello@!" } });
    fireEvent.change(textarea, { target: { value: "Hello" } });
    const warningMessage = screen.queryByText(
      "Special characters are not allowed."
    );
    expect(warningMessage).not.toBeInTheDocument();
  });
});
