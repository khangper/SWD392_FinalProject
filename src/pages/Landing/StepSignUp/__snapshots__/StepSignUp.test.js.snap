// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepSignUp renders the StepSignUp component 1`] = `
<div>
  <div
    class="login-container"
  >
    <div
      class="login-logo-container"
    >
      <a
        href="/home"
      >
        <img
          alt="Cursus Logo"
          class="login-logo"
          src="https://gambolthemes.net/html-items/cursus-new-demo/images/logo.svg"
        />
      </a>
    </div>
    <div
      class="login-form-container"
    >
      <div
        class="tablist"
      >
        <div
          class="tab active"
        >
          <div
            class="instructor-sign-up"
          >
            Instructor Sign Up
          </div>
        </div>
        <div
          class="tab "
        >
          <div
            class="student-sign-up"
          >
            Student Sign Up
          </div>
        </div>
      </div>
      <div
        class="instructor-sign-up-form"
      >
        <p
          class="login-bannerInstructor"
        >
          Sign Up and Create Course!
        </p>
        <form>
          <select
            class="selectpicker"
          >
            <option
              value=""
            >
              Select Category
            </option>
            <option
              value="1"
            >
              Development
            </option>
            <option
              value="2"
            >
              Business
            </option>
            <option
              value="3"
            >
              Finance & Accounting
            </option>
            <option
              value="4"
            >
              IT & Software
            </option>
            <option
              value="5"
            >
              Office Productivity
            </option>
            <option
              value="6"
            >
              Personal Development
            </option>
            <option
              value="7"
            >
              Design
            </option>
            <option
              value="8"
            >
              Marketing
            </option>
            <option
              value="9"
            >
              Lifestyle
            </option>
            <option
              value="10"
            >
              Photography
            </option>
            <option
              value="11"
            >
              Health & Fitness
            </option>
            <option
              value="12"
            >
              Music
            </option>
            <option
              value="13"
            >
              Teaching & Academics
            </option>
          </select>
          <div
            class="ui search focus mt-15"
          >
            <div
              class="ui form swdh30"
            >
              <div
                class="singnupstep-field"
              >
                <textarea
                  class="textarea"
                  id="id_about"
                  name="description"
                  placeholder="Write a little description about you..."
                  rows="3"
                />
              </div>
            </div>
            <div
              class="help-block"
            >
              Your biography should have at least 12000 characters.
            </div>
          </div>
          <button
            class="loginSingUp-btn"
            type="submit"
          >
            Instructor Sign Up Now
          </button>
        </form>
        <p
          class="top38px"
        >
          Already have an account?
           
          <a
            class="LoginMove"
            href="/Login"
          >
            Log In
          </a>
        </p>
      </div>
    </div>
    <div
      class="form-footer"
    >
      <img
        alt=""
        class="footer-logo"
        src="https://gambolthemes.net/html-items/cursus-new-demo/images/sign_logo.png"
      />
      <span>
        Â© 2024 
        <strong>
          Cursus
        </strong>
        . All Rights Reserved.
      </span>
    </div>
  </div>
</div>
`;
